lib/fuse_lowlevel.c

```diff
@@ -1955,10 +1969,10 @@ static void do_init(fuse_req_t req, fuse_ino_t nodeid, const void *inarg)
 			bufsize);
 		bufsize = FUSE_MIN_READ_BUFFER;
 	}
+	se->bufsize = bufsize;
 
-	bufsize -= 4096;
-	if (bufsize < se->conn.max_write)
-		se->conn.max_write = bufsize;
+	if (bufsize - HEADER_SIZE < se->conn.max_write)
```

Comment:
Let's invert this do `se->conn.max_write > [...]`, since we're imposing a limit on max_write here.